### PR TITLE
Adt 60 allergies

### PR DESCRIPTION
**Pull request by t-young31**
## Reason for changes and summary

See https://github.com/inform-health-informatics/emap_planning/issues/56 

## Checklist for submitter & reviewer

- [ ] copy constructor copies the primary key Id of the entity
- [ ] copy constructor copies all other fields of the entity, starting with `super(other)`
- [ ] unique constraints added if appropriate
- [ ] text definition added to columns which could have more than 255 characters
- [ ] foreign key names are in the form `<entityClass>Id`, and no typos
- [ ] during run, glowroot is checked for query speed and indexes added if needed
- [ ] validation run completed and validated

### CODE REVIEW

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * @author Tom Young
+ */
+@Entity
+@Data
+@NoArgsConstructor
+public class AllergenReaction {
+    /**
+     * \brief Unique identifier in EMAP for this conditionSymptom record.
+     * <p>
+     * This is the primary key for the conditionSymptom table.
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:29]: old table name

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long conditionSymptomId;
+
+    /**
+     * \brief Identifier for the ConditionType associated with this record.
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:36]:
```suggestion
     * \brief Identifier for the PatientCondition associated with this record.
```

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     *
+     * This is a foreign key that joins the conditionSymptom table to the patientCondition table.
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:38]: old name, but also probably don't need this entity name in there

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    @ManyToOne
+    @JoinColumn(name = "patientConditionId", nullable = false)
+    private PatientCondition patientConditionId;
+
+    /**
+     * \brief Human readable name for this conditionSymptom.
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:45]: old name

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * \brief Mapping code for the observation from the standardised vocabulary system. Not yet implemented.
+     */
+    private String standardisedCode;
+
+    /**
+     * \brief Nomenclature or classification system used. Not yet implemented.
+     */
+    private String standardisedVocabulary;
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:57]: nah remove it, we've done placeholders for too long 🔥

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:58]: We probably want to know when they have the reactions from (storedFrom and validFrom). The question is, do we want to have an Audit log of them?

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/ConditionType.java [hunk 7]
```diff
@@ -37,22 +37,26 @@ public class ConditionType extends TemporalCore<ConditionType, ConditionTypeAudi
...
-    @Column(nullable = false)
     private String internalCode;
 
     /**
      * \brief Human readable name for this conditionType.
      */
     private String name;
 
+    /**
+     * \brief Subtype of the condition e.g. an allergy to food
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/ConditionType.java:56]: probably worth noting that this is only populated in the context of allergies

**Review by stefpiatek — APPROVED** [review]
🦜

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java (hunk 1 above)
Comment (t-young31) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java:29]: ugh, words are difficult. was going with `AllergyReaction`, if you think that's reasonable

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java (hunk 6 above)
Comment (t-young31) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/conditions/AllergenReaction.java]: I would have said yes, but Anika suggested not to. Perhaps worth a discussion on this

**Review by skeating — APPROVED** [review]

**Pull request merged**